alacritty_terminal/src/term/mod.rs

```diff
@@ -1457,6 +1458,36 @@ impl<T> Term<T> {
             rendered: false,
         }
     }
+
+    fn grid_to_string_between(&self, y0: Line, y1: Line) -> String {
+        let grid = &self.grid;
+        // Use rayon to iterate over each row.
+        (*y0..*y1).into_par_iter().rev().map(|y| {
+            // Create a String and fill it with the contents of the current row.
+            let mut s = String::with_capacity(*grid.cols());
+            let row = &grid[y];
+            for x in 0..*grid.cols() {
+                let cell = &row[Column(x)];
+                s.push(cell.c);
+            }
```

Comment:
This doesn't create the correct output string but instead will lead to a lot of trailing whitespace.

Reply:
It now iterates over each row using `row.occ`. It reduces the amount of trailing whitespace unless something like `.last_mut()`  was called on the row. Now it also calls `.trim_end()` on the output in a different thread.